apps/challenges/views.py

```diff
@@ -506,6 +523,75 @@ def add_participant_team_to_challenge(
+    try:
+        challenge = Challenge.objects.get(pk=challenge_pk)
+    except Challenge.DoesNotExist:
+        response_data = {"error": "Challenge does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    try:
+        participant_team = ParticipantTeam.objects.get(pk=participant_team_pk)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {"error": "Participant Team does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    if (challenge.approved_participant_teams.filter(pk=participant_team_pk).exists()):
+        response_data = {"error": "Participant Team already approved"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+    else:
+        if (ParticipantTeam.objects.filter(team_name__in=challenge.participant_teams.values_list('team_name', flat=True)).exists()):
+            challenge.approved_participant_teams.add(participant_team)
+            response_data = {"success": "Participant Team added to approved list"}
+            return Response(response_data, status=status.HTTP_201_CREATED)
+        else:
+            response_data = {"error": "Participant isn't interested in challenge"}
+            return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+
+@api_view(["POST"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def remove_participant_team_from_approved_list(request, challenge_pk, participant_team_pk):
+    """
+    Remove participant team from approved list
+    """
+    try:
+        challenge = Challenge.objects.get(pk=challenge_pk)
+    except Challenge.DoesNotExist:
+        response_data = {"error": "Challenge does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    try:
+        participant_team = ParticipantTeam.objects.get(pk=participant_team_pk)
+    except ParticipantTeam.DoesNotExist:
+        response_data = {"error": "Participant Team does not exist"}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    all_challenge_phases = ChallengePhase.objects.filter(challenge=challenge)
+    if (all_challenge_phases.count() > 0):
+        for challenge_phase in all_challenge_phases:
+            submission_exist = Submission.objects.filter(participant_team=participant_team, challenge_phase=challenge_phase).exists()
+            if submission_exist:
+                break
+    else:
+        submission_exist = False
+    if challenge.approved_participant_teams.filter(pk=participant_team_pk).exists() and not submission_exist:
+        challenge.approved_participant_teams.remove(participant_team)
+        return Response(status=status.HTTP_204_NO_CONTENT)
+    else:
+        response_data = {"error": "Participant Team not approved or already have a Submission"}
```

Comment:
"Participant Team {name} was not approved"
"Participant Team {name} has existing submissions and cannot be unapproved"

^ Make these cases separate and use this text.

Reply:
Done